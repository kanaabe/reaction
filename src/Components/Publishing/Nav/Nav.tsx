import React from "react"
import Waypoint from "react-waypoint"
import styled, { StyledFunction } from "styled-components"
import { pMedia } from "../../Helpers"
import { Fonts } from "../Fonts"
import { PartnerInline } from "../Partner/PartnerInline"

interface Props extends React.HTMLProps<HTMLDivElement> {
  canFix?: boolean
  sponsor?: any
  title?: string
  transparent?: boolean
}

interface State {
  isFixed: boolean
}

interface DivProps {
  transparent: boolean
  isFixed: boolean
}

export class NavComponent extends React.Component<Props, State> {
  static defaultProps = {
    canFix: true,
    title: "Artsy Editorial",
    transparent: false
  }

  state = {
    isFixed: false
  }

  setPosition = (isFixed) => {
    const { canFix } = this.props
    const currentPosition = this.state.isFixed

    if (canFix && isFixed !== currentPosition) {
      this.setState({ isFixed })
    }
  }

  render() {
    const {
      sponsor,
      className,
      canFix,
      transparent,
      title
    } = this.props
    const { isFixed } = this.state

    return (
      <div>
        <NavContainer
          transparent={!isFixed && transparent}
          className={className}
          isFixed={canFix && isFixed}
        >
          <PartnerInline
            url={sponsor && sponsor.partner_logo_link}
            logo={sponsor && sponsor.partner_condensed_logo}
            color="white"
            margin="0 10px"
          />
          <Title>
            {title}
          </Title>
        </NavContainer>
        <Waypoint
          onEnter={() => this.setPosition(false)}
          onLeave={() => this.setPosition(true)}
        />
      </div>
    )
  }
}

const Div: StyledFunction<DivProps & React.HTMLProps<HTMLDivElement>> = styled.div

const NavContainer = Div`
  background-color: ${props => props.transparent ? "transparent" : "black"};
  border-bottom: 1px solid white;
  ${props => props.transparent && !props.isFixed && `
    position: absolute;
    top: 0;
  `}
  ${props => props.isFixed && `
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  `}
`
export const Nav = styled(NavComponent) `
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 52px;
  width: 100%;
  color: white;
  z-index: 10;
  a {
    z-index: 10;
  }
`
const Title = styled.div`
  ${Fonts.garamond("s17")}
  position: absolute;
  width: 100%;
  text-align: center;
  font-weight: 600;
  ${pMedia.sm`
    display: none;
  `}
`
