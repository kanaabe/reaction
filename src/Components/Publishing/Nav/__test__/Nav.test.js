import { mount } from "enzyme"
import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import { SponsoredArticle } from "../../Fixtures/Articles"
import { Nav, NavComponent } from "../Nav"

describe("Nav", () => {
  it("renders a Nav", () => {
    const nav = renderer.create(<Nav />)
    expect(nav).toMatchSnapshot()
  })
  it("renders a transparent nav", () => {
    const nav = renderer.create(<Nav transparent />)
    expect(nav).toMatchSnapshot()
  })
  it("renders a sponsored nav", () => {
    const nav = renderer.create(<Nav sponsor={SponsoredArticle.sponsor} />)
    expect(nav).toMatchSnapshot()
  })
  it("includes a partner logo in a sponsored nav", () => {
    const nav = mount(<Nav sponsor={SponsoredArticle.sponsor} />)
    expect(nav.html()).toMatch("https://artsy-media-uploads.s3.amazonaws.com/GEu3iYW6CQhbVxsjpOYwZQ%2FGKL_Wort-Bildmarke_negative_rgb+2.png")
  })
  it("renders a custom title", () => {
    const nav = mount(<Nav title="Custom Title" />)
    expect(nav.text()).toMatch("Custom Title")
  })

  it("setPosition sets the state if props.canFix", () => {
    const nav = mount(<NavComponent />)
    nav.instance().setPosition(true)
    expect(nav.state().isFixed).toBe(true)
  })

  it("setPosition does not set state if without props.canFix", () => {
    const nav = mount(<NavComponent canFix={false} />)
    nav.instance().setPosition(true)
    expect(nav.state().isFixed).toBe(false)
  })
})
