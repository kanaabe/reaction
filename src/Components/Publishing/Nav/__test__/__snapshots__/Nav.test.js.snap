// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders a Nav 1`] = `
.c4 {
  font-family: artsy-icons;
  color: white;
  font-size: 32px;
  margin: 0 5px;
  display: inline-block;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 10px;
}

.c2 img {
  max-height: 32px;
  vertical-align: middle;
}

.c2 .file__IconPlus-s100ic91-0 {
  margin: 0 20px;
}

.c2 .c3 {
  margin: 0;
  vertical-align: middle;
}

.c1 {
  background-color: black;
  border-bottom: 1px solid white;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 52px;
  width: 100%;
  color: white;
  z-index: 10;
}

.c0 a {
  z-index: 10;
}

.c5 {
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 17px;
  line-height: 1.1em;
  position: absolute;
  width: 100%;
  text-align: center;
  font-weight: 600;
}

@media (max-width:720px) {
  .c2 img {
    max-height: 24px;
  }

  .c2 .c3 {
    font-size: 24px;
  }

  .c2 .file__IconPlus-s100ic91-0 {
    margin: 0 10px;
  }
}

@media (max-width:720px) {
  .c5 {
    display: none;
  }
}

<div>
  <div
    className="c0 c1"
  >
    <div
      className="PartnerInline c2"
    >
      <a
        href="/"
      >
        <div
          className="c3 c4"
          onClick={undefined}
          style={undefined}
        >
          
        </div>
      </a>
    </div>
    <div
      className="c5"
    >
      Artsy Editorial
    </div>
  </div>
  <span
    style={
      Object {
        "fontSize": 0,
      }
    }
  />
</div>
`;

exports[`Nav renders a sponsored nav 1`] = `
.c4 {
  font-family: artsy-icons;
  color: white;
  font-size: 32px;
  margin: 0 5px;
  display: inline-block;
}

.c6 {
  width: 15px;
  height: 15px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 10px;
}

.c2 img {
  max-height: 32px;
  vertical-align: middle;
}

.c2 .c5 {
  margin: 0 20px;
}

.c2 .c3 {
  margin: 0;
  vertical-align: middle;
}

.c1 {
  background-color: black;
  border-bottom: 1px solid white;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 52px;
  width: 100%;
  color: white;
  z-index: 10;
}

.c0 a {
  z-index: 10;
}

.c7 {
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 17px;
  line-height: 1.1em;
  position: absolute;
  width: 100%;
  text-align: center;
  font-weight: 600;
}

@media (max-width:900px) {
  .c6 {
    width: 13px;
    height: 13px;
  }
}

@media (max-width:720px) {
  .c2 img {
    max-height: 24px;
  }

  .c2 .c3 {
    font-size: 24px;
  }

  .c2 .c5 {
    margin: 0 10px;
  }
}

@media (max-width:720px) {
  .c7 {
    display: none;
  }
}

<div>
  <div
    className="c0 c1"
  >
    <div
      className="PartnerInline c2"
    >
      <a
        href="/"
      >
        <div
          className="c3 c4"
          onClick={undefined}
          style={undefined}
        >
          
        </div>
      </a>
      <svg
        className="c5 c6"
        version="1.1"
        viewBox="0 0 15 15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fillRule="evenodd"
          stroke="none"
          strokeWidth="2"
        >
          <g
            fill="white"
            transform="translate(-101.000000, -18.000000)"
          >
            <g
              transform="translate(101.000000, 18.000000)"
            >
              <rect
                height="1"
                width="15"
                x="0"
                y="7"
              />
              <rect
                height="1"
                transform="translate(7.500000, 7.500000) rotate(-90.000000) translate(-7.500000, -7.500000) "
                width="15"
                x="0"
                y="7"
              />
            </g>
          </g>
        </g>
      </svg>
      <a
        href="http://artsy.net"
        onClick={[Function]}
        target="_blank"
      >
        <img
          src="https://artsy-media-uploads.s3.amazonaws.com/eZqsrnodcIlyyJzmRqLm4A%2FBombay_Sapphire_logo.png"
        />
      </a>
    </div>
    <div
      className="c7"
    >
      Artsy Editorial
    </div>
  </div>
  <span
    style={
      Object {
        "fontSize": 0,
      }
    }
  />
</div>
`;

exports[`Nav renders a transparent nav 1`] = `
.c4 {
  font-family: artsy-icons;
  color: white;
  font-size: 32px;
  margin: 0 5px;
  display: inline-block;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 10px;
}

.c2 img {
  max-height: 32px;
  vertical-align: middle;
}

.c2 .file__IconPlus-s100ic91-0 {
  margin: 0 20px;
}

.c2 .c3 {
  margin: 0;
  vertical-align: middle;
}

.c1 {
  background-color: transparent;
  border-bottom: 1px solid white;
  position: absolute;
  top: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 52px;
  width: 100%;
  color: white;
  z-index: 10;
}

.c0 a {
  z-index: 10;
}

.c5 {
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 17px;
  line-height: 1.1em;
  position: absolute;
  width: 100%;
  text-align: center;
  font-weight: 600;
}

@media (max-width:720px) {
  .c2 img {
    max-height: 24px;
  }

  .c2 .c3 {
    font-size: 24px;
  }

  .c2 .file__IconPlus-s100ic91-0 {
    margin: 0 10px;
  }
}

@media (max-width:720px) {
  .c5 {
    display: none;
  }
}

<div>
  <div
    className="c0 c1"
  >
    <div
      className="PartnerInline c2"
    >
      <a
        href="/"
      >
        <div
          className="c3 c4"
          onClick={undefined}
          style={undefined}
        >
          
        </div>
      </a>
    </div>
    <div
      className="c5"
    >
      Artsy Editorial
    </div>
  </div>
  <span
    style={
      Object {
        "fontSize": 0,
      }
    }
  />
</div>
`;
